x/warp/README.md

```diff
@@ -0,0 +1,190 @@
+4. Filter the validators of Subnet B to include only the validators that the signed message claims as signers
+5. Verify the claimed included validators represent a sufficient quorum of stake to verify the message
+6. Aggregate the BLS Public Keys of the claimed signers into an aggregated BLS Public Key
+7. Validate the aggregate signature matches the claimed aggregate BLS Public Key
+
+This means that if Subnet B has 10 validators, then C-Chain to Subnet communication only requires a threshold of those 10 validators rather than the entire Primary Network.
+
+Since the security of Subnet B depends on the validators of Subnet B already, changing the requirements of verifying the message from verifying a signature from the entire Primary Network to only Subnet B's validator set does not change the security of Subnet B!
+
+## Warp Precompile
+
+The Warp Precompile is broken down into three functions defined in the Solidity interface file [here](../../../contract-examples/contracts/IWarpMessenger.sol).
+
+### sendWarpMessage
+
+`sendWarpMessage` is used to send a verifiable message. Calling this function results in sending a message with the following contents:
+
+- `SourceChainID` - blockchainID of the sourceChain on the Avalanche P-Chain
+- `SourceAddress` - `msg.sender` encoded as a 32 byte value that calls `sendWarpMessage`
+- `DestinationChainID` - `bytes32` argument specifies the blockchainID on the Avalanche P-Chain that should receive the message
+- `DestinationAddress` - 32 byte value that represents the destination address that should receive the message (on the EVM this is the 20 byte address left zero extended)
+- `Payload` - `payload` argument specified in the call to `sendWarpMessage`
+
+Calling this function will issue a `SendWarpMessage` event from the Warp Precompile. Since the EVM limits the number of topics to 4 including the EventID, this message includes only the topics that would be expected to help filter messages emitted from the Warp Precompile the most.
+
+Specifically, the `payload` is not emitted because this would need to be a hash and it would not be expected to be helpful to determine whether or not the log is of any interest to anyone parsing the chain.
+
+Additionally, the `SourceChainID` is excluded because anyone parsing the chain can be expected to already know the blockchainID. Therefore, the `SendWarpMessage` event includes the indexable attributes:
+
+- `destinationChainID`
+- `destinationAddress`
+- `sender`
+
+The actual `message` is the entire [Avalanche Warp Unsigned Message](https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/warp/unsigned_message.go#L14) including the Subnet-EVM [Addressed Payload](../../../warp/payload/payload.go).
+
+
+### getVerifiedMessage
+
+`getVerifiedMessage` is used to read the contents of the delivered Avalanche Warp Message into the expected format.
+
+It returns the message if present and a boolean indicating if a message is present.
```

Comment:
Does this mean getVerifiedMessage returns only if there is an accepted warp message in the same block? How do other contracts can use this to see if a particulat message was accepted?